src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -556,12 +561,22 @@ std::unique_ptr<AggregatingProjectionStep> AggregatingStep::convertToAggregating
     return aggregating_projection;
 }
 
+void AggregatingStep::updateParams(const Names & new_keys, const std::unordered_map<std::string, std::string> & changed_to_new_key_mapping)
+{
+    params.keys = new_keys;
```

Comment:
that is very slippery road. we store argument column names in many places, e.g. [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/AggregateDescription.h#L17) and [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/Aggregator.h#L1269). why it works now btw?)
I think it might be safer to alias columns on themselves. like `select x - 100 as x`. I remember we discussed it some time ago, could you pls remember what we come up with?